openhands/storage/file_settings_store.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import annotations
+
+import json
+from dataclasses import dataclass
+
+from openhands.core.config.app_config import AppConfig
+from openhands.server.settings import Settings
+from openhands.storage import get_file_store
+from openhands.storage.files import FileStore
+from openhands.storage.settings_store import SettingsStore
+
+
+@dataclass
+class FileSettingsStore(SettingsStore):
+    file_store: FileStore
+    path: str = 'settings.json'
+
+    async def load(self) -> Settings | None:
+        try:
+            json_str = self.file_store.read(self.path)
+            kwargs = json.loads(json_str)
+            settings = Settings(**kwargs)
+            return settings
+        except FileNotFoundError:
+            return None
+
+    async def store(self, settings: Settings):
+        json_str = json.dumps(settings.__dict__)
+        self.file_store.write(self.path, json_str)
+
+    @classmethod
+    async def get_instance(cls, config: AppConfig, token: str | None):
+        file_store = get_file_store(config.file_store, config.file_store_path)
```

Comment:
I wonder if we should just have a singleton file_store. We do this in `openhands/server/shared`....

Reply:
We probably want a singleton of this class too...

I'm creating a `ConversationStore`, which now that I think about it is kind of silly (why give ourselves 10 classes to override instead of 1)...